selectors/emojis.ts

```diff
@@ -35,31 +37,54 @@ export const getRecentEmojis = createSelector(
             '[]',
         );
     },
-    (recentEmojis: string) => {
+    getUserSkinTone,
+    (recentEmojis: string, userSkinTone: string) => {
         if (!recentEmojis) {
             return [];
         }
+
         const parsedEmojiData: RecentEmojiData[] = JSON.parse(recentEmojis);
-        return parsedEmojiData;
+        return normalizeRecentEmojisData(parsedEmojiData, userSkinTone);
     },
 );
 
+export function normalizeRecentEmojisData(data: RecentEmojiData[], userSkinTone: string) {
```

Comment:
This is now the important part of this PR where it dedupes and "fixes" the skin tone of the recent emojis when they're pulled out of preferences. I was a bit worried about doing this every time the preferences change, but I don't think it should be a huge deal in practice. It'll also only recalculate when the user's recent emojis or skin tone change which hopefully shouldn't happen too often